### PR TITLE
[Upgrade for DLFolderName]Lps 182296 followup

### PR DESCRIPTION
**Pull request by slnn**
Hi @yuhai 

 I dig into the process that how portal will handle these DLFolders when uploading a portrait for a user and displaying the portrait.

I found out that these target DLFolders which need to update the value of their name in the DB should satisfy these condition

1>The DLFolder used  "java.lang.Class" as name
2>Its parent Folder's name is the current user's userId
3>Its root Folder(whose parentFolderId is 0) is a hidden folder

By the way, the root folder will never be a hidden folder if it is created from UI.

Thanks
Lily

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:lima

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/upgrade/v1_0_1/DLFolderNameUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+
+	private List<Long> _getUserIds() {
+		List<Long> userIds = new ArrayList<>();
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"select userId from User_ where portraitId != 0")) {
+
+			ResultSet resultSet = preparedStatement.executeQuery();
```
Comment (yuhai) [inline modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/upgrade/v1_0_1/DLFolderNameUpgradeProcess.java:92]: @slnn, the case **portraitId != 0** will miss the case:

1. Use empty database and as Test user login.
2. Go to Test User -> My profile -> Edit
3. Change Image, select the image. 
Check the related dlFolder data will be created and user_.portraitId is 0.

**Comment by yuhai** [comment]
@slnn, I checked the process of uploading image for user profile, please see the below sql if it can get the right data.
```
select dlfolder.folderId from dlfolder inner join repository on dlfolder.repositoryId = repository.repositoryId inner join group_ on repository.groupId = group_.groupId inner join user_  on group_.classPK = user_.userId where dlfolder.name = 'java.lang.Class'
```

If this can satisfy with the case. Please use the sql to get.

Thanks,
Hai

**Comment by slnn** [comment]
Please see https://github.com/liferay-core-infra/liferay-portal/pull/4664

**Pull request closed without merging**